lib/Sema/ConstraintSystem.h

```diff
@@ -1139,6 +1140,9 @@ class ConstraintSystem {
   using FixedRequirement = std::tuple<TypeBase *, RequirementKind, TypeBase *>;
   SmallVector<FixedRequirement, 4> FixedRequirements;
 
+  /// Intended to track information about type variable bindings.
+  llvm::DenseMap<const TypeVariableType *, const TypeVariableBinding *> TypeVariableBindings;
```

Comment:
The problem with doing it this way I see is related to the amount of memory we are going to spent in tracking, we already have a way to track bindings in the type variable so I attaching locator information to that should be less expensive then keeping a separate data structure to keep track of bindings...

Reply:
@xedin what would be this way you mention that the bindings are already tracked in the type variable? 
I'm struggling for a while now with how to get the locator for the source of the fixed type ... and also how to get this into `assignFixedType`